ladybug/hourlyplot.py

```diff
@@ -0,0 +1,424 @@
+
+from .analysisperiod import AnalysisPeriod
+from .datacollection import HourlyContinuousCollection, HourlyDiscontinuousCollection
+from .graphic import GraphicContainer
+from .legend import LegendParameters
+
+from ladybug_geometry.geometry2d.pointvector import Point2D, Vector2D
+from ladybug_geometry.geometry2d.line import LineSegment2D
+from ladybug_geometry.geometry2d.polygon import Polygon2D
+from ladybug_geometry.geometry2d.polyline import Polyline2D
+from ladybug_geometry.geometry2d.mesh import Mesh2D
+from ladybug_geometry.geometry3d.pointvector import Point3D, Vector3D
+from ladybug_geometry.geometry3d.line import LineSegment3D
+from ladybug_geometry.geometry3d.plane import Plane
+from ladybug_geometry.geometry3d.polyline import Polyline3D
+from ladybug_geometry.geometry3d.mesh import Mesh3D
+
+
+class HourlyPlot(object):
+    """Module for visualization of houlry data collections.
+
+    Args:
+        data_collection: A HourlyContinuousCollection or HourlyDiscontinuousCollection
+            which will be used to generate the hourly plot.
+        legend_parameters: An optional LegendParameter object to change the display
+            of the HourlyPlot (Default: None).
+        base_point: An optional Point3D to be used as a starting point to generate
+            the geometry of the plot (Default: (0, 0, 0)).
+        x_dim: An optional number to set the X dimension of the mesh cells (Default: 1).
+        y_dim: An optional number to set the Y dimension of the mesh cells (Default: 4).
+        z_dim: An optional number to set the Z dimension of the entire chart. This
+            will be used to make the colored_mesh3d of the chart vary in the Z
+            dimension according to the data. The value input here should usually be
+            several times larger than the x_dim or y_dim in order to be noticable
+            (e.g. 100). If 0, the colored_mesh3d will simply be flat. (Default: 0).
+
+    Properties:
+        * data_collection
+        * legend_parameters
+        * base_point
+        * x_dim
+        * y_dim
+        * z_dim
+        * colored_mesh2d
+        * colored_mesh3d
+        * legend
+        * chart_border2d
+        * chart_border3d
+        * hour_lines2d
+        * hour_lines3d
+        * hour_label_points2d
+        * hour_label_points3d
+        * hour_labels
+        * month_lines2d
+        * month_lines3d
+        * month_label_points2d
+        * month_label_points3d
+        * month_labels
+        * title_text
+        * lower_title_location
+        * upper_title_location
+        * analysis_period
+        * values
+        * value_colors
+    """
+    __slots__ = ('_data_collection', '_base_point', '_x_dim', '_y_dim', '_z_dim',
+                 '_num_y', '_num_x', '_container', '_colored_mesh2d', '_colored_mesh3d',
+                 '_hour_points', '_hour_text', '_month_points', '_month_label_points',
+                 '_month_text')
+
+    HOUR_LABELS = (0, 6, 12, 18, 24)
```

Comment:
I think this will be something that people want to have access to. Specially now that you are providing them a very well-designed plot with a lot of flexibility.

Reply:
That's why I made it public. If people really want to change it everywhere, they can overwrite `HOUR_LABELS `. But I'll throw in a method to this class that can generate custom hour lines and text labels based on any list of labels that the user inputs. This should satisfy your comment here.

Reply:
Done!